### ISSUE TITLE
ability to always check files in a directory

### ISSUE THREAD

**Issue by pcnate** [yokawasa/action-sqlcheck#16]
I have a build that creates an installation script from multiple input files that were generated. I do not seen an option to specify the folder. This folder is in my .gitignore. Anything like this exist?

**Comment by yokawasa** [comment]
@pcnate 

There is no option to specify the folder now, but i think it's a good option so I would like to add it as a new feature.

How do you like specifying the folder with the following `directories` parameter?  If you have multiple folder, maybe you can specify them with comma separator like `directories: resources/mydir1,resources/mydir2,resources/mydir3`

```yaml
jobs:
  sqlcheck:
    name: sqlcheck job
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@master
    - uses: yokawasa/action-sqlcheck@v1.2.1
      with:
        post-comment: true
        directories: resources/mydir
        token: ${{ secrets.GITHUB_TOKEN }}
```

By the way, if your folder is in your .gitignore, files under the folder will not be pushed into your GitHub repo, which mean the GitHub action cannot check them.

**Comment by pcnate** [comment]
Would it be better to use multiple lines in that case? I'm not great with yaml structure yet but something like this?

```yaml
jobs:
  sqlcheck:
    name: sqlcheck job
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@master
    - uses: yokawasa/action-sqlcheck@v1.2.1
      with:
        post-comment: true
        directories:
        - directory: resources/mydir1
        - directory: resources/mydir2
        token: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by yokawasa** [comment]
@pcnate 
Thanks for the idea! Using multi-lines definitely look better but unfortunately array isn't supported type for action inputs

> Action inputs only supports string keys and string values at this time. This is because they translate to environment variables within the action’s execution run.
( ref: https://github.community/t/can-action-inputs-be-arrays/16457 )

But as suggested in page above, multi-line string can be possible. 
So how about this?

```
jobs:
  sqlcheck:
    name: sqlcheck job
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@master
    - uses: yokawasa/action-sqlcheck@v1.2.1
      with:
        post-comment: true
        directories: |
          resources/mydir1
          resources/mydir2
        token: ${{ secrets.GITHUB_TOKEN }}
```
By default, the action checks and identifies anti-patterns in SQL queries in any files in PR queries. But you can filter target files under specific directory(s) by specifying them using `directories` input parameter

**Comment by pcnate** [comment]
Oh that looks even cleaner

**Comment by pcnate** [comment]
My process involves compiling together many SQL files as well as JSON files that contain SQL into a deployment SQL script in a build folder that is not part of the repository. If actions were able to check any files that were changed as one step, and then after the build check the output files, that would be excellent.

**Comment by yokawasa** [comment]
i see. My understanding is that in your scenario `directories` parameter is NOT for filtering files in PR queries but for checking any files under the directories whether they are part of the repo or not. Correct?

**Comment by yokawasa** [comment]
@pcnate I'm going to add the following feature in the next version:

- Add input paramter `directories` that allow you to add path(s) of directory under which the action check any files whether they are part of the repository or not. By default, the action checks only files in PR queries 
- Add output variable `issue-found` that is a boolean value to indicate an issue was found in the files that sqlcheck action checked

let me know if it'll work for your scenario

**Comment by yokawasa** [comment]
I've just published v1.3.0 where the following features are added:
- Add input paramter `directories` that allow you to add path(s) of directory under which the action check any files whether they are part of the repository or not. By default, the action checks only files in PR queries - [issue #16](https://github.com/yokawasa/action-sqlcheck/issues/16)
- Add output variable `issue-found` that is a boolean value to indicate an issue was found in the files that sqlcheck action checked

- https://github.com/yokawasa/action-sqlcheck/pull/17
- https://github.com/yokawasa/action-sqlcheck/releases/tag/v1.3.0

@pcnate 
I'm closing the issue.
Please open an issue if you have further discussion or suggestion!
Thanks again for the issue.
